### DISCUSSION TITLE
About Diffusers

### DISCUSSION THREAD

**Discussion by GrimorJander** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#319] [General]
Hi people.
In order to sharpen the pixels, and reduces the glare, i´m thinking about to use some kind of filter / difusser on the RGB led matrix.
Some people (it was @witnessmenow ?) say a a plain white sheet of paper will do the trick.
Others (Lady Ada) sell a specific solution of [Black LED Diffusion Acrylic Panel](https://www.adafruit.com/product/4594).
I´m looking for the cheapest as possible, like onion paper sheet or something like that. 
What do you think will be the best option?

**Comment by lmurdock12** [comment]
Find any solutions?

**Reply by jlarnal** [reply]
There's also the solder stencil as an aperture mask, as seen here : https://www.youtube.com/watch?v=oLgUtjyKO6Q 👍

**Comment by DarrylStrong** [comment]
I have used a vinyl applied to a sheet of clear/tinted acrylic in the past. It will require experimentation to find the look you require.

**Comment by DarrylStrong** [comment]
On our captioning signs we use a coated dark grey acrylic from Charvo (UK) to offer some anti-reflective properties as well as the increased contrast a dark acrylic offers. Be wary though as the distance of the LEDs to the diffuser really affects the result so keeping this consistent across the display can be troublesome (we manufacture signs which are 384x64 at 4mm pitch and 352x64 at 5mm).

**Comment by GrimorJander** [comment]
There really is no single solution.
It depends a little on the needs of each viewing place.
I have tried different diffusers and filters, neutral density, polarized... and I think that the most successful is the Adafruit solution. The slightly black tinted glass helps reduce excess pixel glare and increases the perception of sharpness especially at short viewing distances.
 I think a cheap tinted auto glass adhesive could be a cheap solution to bonding regular glass.
![IMG_20220930_120937](https://user-images.githubusercontent.com/27100419/193249270-a4e65524-c9e5-4284-8cad-c754282b555c.jpg)

**Comment by mrcodetastic** [comment]
I tried both 2mm and 3mm thick white opaque acrylic (Perspex) and it was far too blurred.

**Comment by DarrylStrong** [comment]
Here is one of our signs fitted with Perspex 923 (dark grey) which is anti reflective coated. P2.5 384x64 display.

https://photos.app.goo.gl/bG7DpsJiShio8Smw5

**Reply by mrcodetastic** [reply]
What thickness of acrylic?  This looks much better. I have a small 64x32 panel as my desk clock, but it has always been too glary and never found a decent solution other than turning the brightness down. This seems to make it bright and less eye-burning.

**Reply by DarrylStrong** [reply]
3mm, I have a case design too.
[]![Uploading c{8809da1} (5f2bd84d-542b-4ca2-8467-55d6c080865f id=0x7f0a03cc tag=BaseCommentFragment) …]

That's a P2.5 panel.

**Comment by DarrylStrong** [comment]
And another shot...

![IMG_20210429_095610](https://user-images.githubusercontent.com/48095813/193419504-206bdef0-ed76-46df-8d1c-3f5809a0140a.jpg)

**Reply by mrcodetastic** [reply]
Glad to see this library being used to create such a great product in the UK (living in GBR myself).

Out of curiosity, I have a bunch of p3 panels I want to join together, do you have metal brackets designed I could use? I don't want a whole box or anything - something that's like a O==O bit of metal that can join these up side by side with the screw holes.

**Reply by GrimorJander** [reply]
I'm using a custom DIY frame with 25mm X 25mm right angle profile of aluminium. https://shop.mtcss.co.uk/product/25mm-x-25mm-ali-angle/
I cost me 5 euros a long piece of 2 meters. 
I cut it 4 piezes of total panel lenght and with and the tricky part is drill holes in the exact place for the m3 screws. The end product is a perfect big panel, where you cannot see the joining. I think a frame is better than rear joining.

**Reply by DarrylStrong** [reply]
I get the metalwork made by our local sheet metal workers. All cnc laser cut etc.


I do use our cnc mill for cutting perspex to make up demo/test plates though.

I am in Hamble near Southampton, if you are local I could machine something up for you to collect?

**Comment by board707** [comment]
@DarrylStrong 
What is the color depth and what FPS do you get on a panel of this size?

**Reply by DarrylStrong** [reply]
Colour depth of three, only displaying text so no issue with contouring on colours.

Mostly static text but I have implemented scrolling by bit manipulation of the display's DMA store.

As for refresh rate, I will have to report back when in the office.

**Reply by GrimorJander** [reply]
Looks great with the acrylic.
I wish I buy 2.5P instead P4 because my viewing distance is really short (1.5m)

**Reply by board707** [reply]
> Colour depth of three, only displaying text so no issue with contouring on colours.
> Mostly static text 

Thank for your answer.
For colour depth of three and static text, you can double the size of your sign, I think

**Reply by DarrylStrong** [reply]
I have tested 512x64 and that worked well. It is amazing what such a small device can drive!

**Comment by mzashh** [comment]
My solution is to use a grid and then tracing paper, and clear acrylic or glass, or you can also tint the glass.

the grid i use is customisable for all kind of panel pitches and resolutions and cam be printed in seperate pieces 

https://www.thingiverse.com/thing:4216443

as for the case i have only used single panels for now so i use this adafruit case which i modify as per my need, by default its for a 32x32 p6 panel

https://learn.adafruit.com/32x32-square-pixel-display/3d-printing
